### PR TITLE
moving to upstash and new integration

### PR DESCRIPTION
**Pull request by noahfschr**

### CODE REVIEW

File: examples/with-redis/README.md [hunk 1]
```diff
@@ -1,96 +1,65 @@
-# Redis Example
+# Roadmap Voting App 
 
-This example showcases how to use Redis as a data store in a Next.js project. [Lambda Store](https://lambda.store/) is used as managed Redis service.
+<br/>
```
Comment (leerob) [inline examples/with-redis/README.md:3]:
```suggestion
```

File: examples/with-redis/README.md [hunk 2]
```diff
@@ -1,96 +1,65 @@
-# Redis Example
+# Roadmap Voting App
```
Comment (leerob) [inline examples/with-redis/README.md:1]:
```suggestion
# Redis Example (with Upstash)
```

File: examples/with-redis/README.md [hunk 3]
```diff
@@ -1,96 +1,65 @@
...
-npx create-next-app --example with-redis with-redis-app
-# or
-yarn create next-app --example with-redis with-redis-app
-```
+See
+[https://roadmap.upstash.com](https://roadmap.upstash.com)
+                                    
+## Deploy Your Own
+You can deploy Roadmap Voting App for your project/company using [Vercel and Upstash](https://vercel.com/integrations/upstash) clicking the below button:
+<br/>
```
Comment (leerob) [inline examples/with-redis/README.md:17]: You shouldn't need any line breaks with Markdown
```suggestion
```

File: examples/with-redis/README.md [hunk 4]
```diff
@@ -1,96 +1,65 @@
...
+<br/>
```
Comment (leerob) [inline examples/with-redis/README.md:18]:
```suggestion
```

File: examples/with-redis/package.json [hunk 5]
```diff
@@ -1,18 +1,17 @@
 {
-  "name": "with-redis",
+  "name": "roadmap-voting",
```
Comment (leerob) [inline examples/with-redis/package.json:2]:
```suggestion
  "name": "with-redis",
```

Let's keep this the same as the folder name

File: examples/with-redis/package.json [hunk 6]
```diff
@@ -1,18 +1,17 @@
...
   "version": "0.1.0",
+  "private": true,
```
Comment (leerob) [inline examples/with-redis/package.json:4]:
```suggestion
```

File: examples/with-redis/package.json [hunk 7]
```diff
@@ -1,18 +1,17 @@
...
   "scripts": {
     "dev": "next dev",
     "build": "next build",
     "start": "next start"
   },
   "dependencies": {
-    "next": "latest",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
+    "next": "10.0.6",
```
Comment (leerob) [inline examples/with-redis/package.json:11]:
```suggestion
    "next": "latest",
```

File: examples/with-redis/pages/api/create.js [hunk 8]
```diff
@@ -1,36 +1,25 @@
...
-      body: 'success',
-    })
-  } else {
-    client.quit()
-    res.json({
-      error: 'Feature can not be empty',
-    })
-  }
+module.exports = async (req, res) => {
+    let redis = new Redis(fixUrl(process.env.REDIS_URL));
```
Comment (leerob) [inline examples/with-redis/pages/api/create.js:5]: Why don't we set up Redis in a util file so we don't even need to "fix URL" and we instead import the redis client?

File: examples/with-redis/pages/api/addemail.js [hunk 9]
```diff
@@ -1,37 +1,31 @@
...
-    res.json({
-      body: 'success',
-    })
-  } else {
-    client.quit()
-    res.json({
-      error: 'Invalid email',
-    })
-  }
+    if (email && validateEmail(email) ) {
```
Comment (leerob) [inline examples/with-redis/pages/api/addemail.js:14]: Could we use the same Prettier settings as before? Maybe add to the package.json?

File: examples/with-redis/pages/index.js [hunk 10]
```diff
@@ -1,168 +1,173 @@
...
-  function handleNewFeature(event) {
-    const requestOptions = {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({ title: inputNewFeature.current.value }),
+import { ToastContainer, toast } from 'react-toastify';
+import * as React from "react";
+
+
+class Home extends React.Component {
```
Comment (leerob) [inline examples/with-redis/pages/index.js:6]: Let's not use class components 👍 Could you change this to use hooks?

File: examples/with-redis/pages/index.js [hunk 11]
```diff
@@ -1,168 +1,173 @@
...
+            )
+    }
+
+    vote(event, title) {
+        const requestOptions = {
+            method: 'POST',
+            headers: {'Content-Type': 'application/json'},
+            body: JSON.stringify({"title": title})
+        };
+        console.log(requestOptions);
```
Comment (leerob) [inline examples/with-redis/pages/index.js:49]: You can remove console logs 👍

File: examples/with-redis/README.md [hunk 12]
```diff
@@ -1,96 +1,62 @@
...
-npx create-next-app --example with-redis with-redis-app
-# or
-yarn create next-app --example with-redis with-redis-app
-```
+See
+[https://roadmap.upstash.com](https://roadmap.upstash.com)
+                                    
+## Deploy Your Own
+You can deploy Roadmap Voting App for your project/company using [Vercel and Upstash](https://vercel.com/integrations/upstash) clicking the below button:
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2Fupstash%2Fserverless-tutorials%2Ftree%2Fmaster%2Froadmap-voting-demo&env=LOGO&envDescription=Enter%20URL%20for%20your%20project%2Fcompany%20logo&envLink=https%3A%2F%2Fdocs.upstash.com%2Fdocs%2Ftutorials%2Froadmap_voting_app&project-name=roadmap-voting&repo-name=roadmap-voting&demo-title=Roadmap%20Voting&demo-description=Roadmap%20Voting%20Page%20for%20Your%20Project&demo-url=https%3A%2F%2Froadmap.upstash.com&integration-ids=oac_V3R1GIpkoJorr6fqyiwdhl17)
```
Comment (leerob) [inline examples/with-redis/README.md:16]:
```suggestion
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2Fupstash%2Fserverless-tutorials%2Ftree%2Fmaster%2Froadmap-voting-app&env=LOGO&envDescription=Enter%20URL%20for%20your%20project%2Fcompany%20logo&envLink=https%3A%2F%2Fdocs.upstash.com%2Fdocs%2Ftutorials%2Froadmap_voting_app&project-name=roadmap-voting&repo-name=roadmap-voting&demo-title=Roadmap%20Voting&demo-description=Roadmap%20Voting%20Page%20for%20Your%20Project&demo-url=https%3A%2F%2Froadmap.upstash.com&integration-ids=oac_V3R1GIpkoJorr6fqyiwdhl17)
```

This URL wasn't working 👍

**Comment by noahfschr** [comment]
@leerob resolved your comments. can you review again please?

**Comment by leerob** [comment]
@noahfschr looks like linting failed due to formatting. Could you ensure `yarn lint` passes?

**Review by leerob — DISMISSED** [review]

**Comment by leerob** [comment]
Thank you @noahfschr! 🙌

**Comment by leerob** [comment]
Still an issue with `yarn lint` and then we can merge.

**Review by leerob — DISMISSED** [review]

**Comment by noahfschr** [comment]
@leerob It passes now. Thanks for the review.

**Review by leerob — APPROVED** [review]

**Pull request merged**